ERCS/erc-7524.md

```diff
@@ -0,0 +1,211 @@
+
+One specific example of how such a signature can lead to unique pseudonymity is that we prove it was generated correctly in a ZK-SNARK that only reveals publicly the hash(signature), and the SNARK additionally proves some property the public key has (i.e. is in some anonymity set, has executed some set of actions on chain, etc). This proof is the only thing that is ever seen by other people, and so the hash(signature) can be used as a “nullifier”: a public commitment to a specific anonymous account, to forbid actions like double spending, or allow a consistent identity between anonymous actions. We aim to standardize a new verifiably deterministic signature algorithm that both uniquely identifies the keypair, and keeps the account identity secret, where verification does not require a secret key. The specific signature function is $hash(message, public\ key) ^ {secret\ key}$.
+
+## Motivation
+
+- Existing ZK applications have the advantage that there is no uniqueness constraint on the provers: that is, allowing the same wallet to prove itself as a member more than once is intended. However, many applications require a maximum of one action per user, especially protocols that desire Sybil resistance. Such protocols are not natively possible on Ethereum right now without mapping each address into an opt-in mapping that also maps a user’s private key to a new system, which adds complexity, loses atomicity, and does not benefit from the rich on-chain history of Ethereum accounts.
+- Specific applications that require this tech include:
+    - zk voting, where each account in some set has one vote
+    - pseudonymously claiming an airdrop like Stealthdrop
+    - moderating a pseudonymous forum, where people can prove that they are the same identity elsewhere in the forum
+    - zk proof of solvency — if you want two exchanges to prove they know a set of private keys that hold some balance, you need a way to ensure that two exchanges aren’t both claiming the same address, while keeping it private
+    
+As such, a deterministic value based on the Ethereum account’s ECDSA keypair is a necessary component of ensuring one action per user and enables all these applications on Ethereum.
+    
+
+## Specification
+
+We propose a new signature standard that offers the following properties, to be implemented for standard ECDSA keys within wallets:
+
+1. It produces signatures that contain a deterministic component and a nondeterministic component. The deterministic component may be used as a *nullifier*.
+2. Signers can use existing secpk256k1 keypairs, such as those in hardware wallets that support Ethereum accounts. As a consequence, secret keys can remain in secure enclaves if there is a generator point multiplication API into the enclave (which Ledger for instance has).
+
+### Parameters
+
+This scheme uses the secp256k1 curve, defined in SEC 2: Recommended Elliptic Curve Domain Parameters v2, page 9.
```

Comment:
We don't permit external links (except [as documented in EIP-1](https://eips.ethereum.org/EIPS/eip-1#linking-to-external-resources).)

This particular PDF has a very permissive license, so I'd recommend copying it into your assets directory (`../assets/erc-7542/`) and linking to it there.

Reply:
```
* At ./_site/EIPS/eip-7524.html:252:

  internally linking to ../assets/erc-7524/sec2-v2.pdf, which does not exist

* At ./_site/EIPS/eip-7524.html:281:

  internally linking to ../assets/erc-7524/sec1-v2.pdf, which does not exist
```
  
I get this issue from HTMLProofer when doing that.